text/0000-range-get-and-convert-methods.md

```diff
@@ -0,0 +1,397 @@
+3. A new way to obtain a value from a range based on the `Option` type:
+when `Some` is returned, a value returned is guaranteed to lie
+within the range, and it can't change. As opposed to using the
+`contains()` method, this allows the developer to work with a type
+having a guarantee that this value can't be changed and lies within
+the scope of allowed values by the range, and this fact can't be
+forgotten or abused in the code. It also brings a slightly more
+convenient way of getting a value from the range. Consider a use-case
+when a `Result` type is used. Now it is possible to use the `try!` macro
+or the "question-mark" operator `?` to quickly exit the function when a
+value doesn't lie within the range:
+
+    ```rust
+    fn set_thread_priority(priority: u8) -> Result<(), &'static str> {
+        let value = (0..100)
+            .get_value(&priority)
+            .ok_or_else(|| "The priority doesn't lie within the user-allowed range")?;
+
+        // The same value but mapped to the allowed values range for
+        // the niceness:
+        let mapped = value.convert(&(-20..20))
+            .ok_or_else(|| "The priority doesn't lie within the niceness range")?;
+
+        set_niceness_for_current_thread(*mapped)
+    }
+    ```
+
+<!--
+- Introducing new named concepts.
+- Explaining the feature largely in terms of examples.
+- Explaining how Rust programmers should *think* about the feature, and how it should impact the way they use Rust. It should explain the impact as concretely as possible.
+- If applicable, provide sample error messages, deprecation warnings, or migration guidance.
+- If applicable, describe the differences between teaching this to existing Rust programmers and new Rust programmers.
+- Discuss how this impacts the ability to read, understand, and maintain Rust code. Code is read and modified far more often than written; will the proposed feature make code easier to maintain?
+
+For implementation-oriented RFCs (e.g. for compiler internals), this section should focus on how compiler contributors should think about the change, and give examples of its concrete impact. For policy RFCs, this section should provide an example-driven introduction to the policy, and explain its impact in concrete terms.
+-->
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+<!--
+This is the technical portion of the RFC. Explain the design in sufficient detail that:
+
+- Its interaction with other features is clear.
+- It is reasonably clear how the feature would be implemented.
+- Corner cases are dissected by example.
+
+The section should return to the examples given in the previous section, and explain more fully how the detailed proposal makes those examples work.
+-->
+
+TODO
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+<!-- Why should we *not* do this? -->
+No known and reasonable drawbacks.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
```

Comment:
I think this is missing a section about why this needs to be in `core` instead of a trait, or why it's an important enough ecosystem primitive to be centralized into `core` instead of just letting people use it tomorrow in a crate.